lrtable/src/lib/statetable.rs

```diff
@@ -414,12 +414,10 @@ where
         let off = (usize::from(stidx) * usize::from(self.rules_len)) + usize::from(ridx);
         // Goto entries are encoded by adding 1 to their value, while 0 is reserved for no entry
         // (i.e. error)
-        if self.gotos.get(off).unwrap() == 0 {
-            None
-        } else {
-            // gotos can only contain state id's which we know can fit into StIdxStorageT so this
-            // cast is safe
-            Some(StIdx((self.gotos.get(off).unwrap() - 1) as StIdxStorageT))
+        match self.gotos.get(off) {
+            Some(0) => None,
+            Some(i) => Some(StIdx((i - 1) as StIdxStorageT)),
```

Comment:
Can you get the comment (about the cast) back in here? I like to be explicit about casts, because they're so dangerous.